app/utilities/mappings.py

```diff
@@ -1,10 +1,12 @@
 from typing import Generator, Iterable, Mapping, Sequence
 
+from ordered_set import OrderedSet
+
 
 def get_flattened_mapping_values(
     map_to_flatten: Mapping[tuple, Iterable[str]] | Mapping[str, Iterable[str]]
-) -> set[str]:
-    return {x for v in map_to_flatten.values() for x in v}
+) -> OrderedSet[str]:
+    return OrderedSet({x for v in map_to_flatten.values() for x in v})
```

Comment:
Initialising an ordered set with a normal set doesn't preserve the order, the order is lost before order set can. Do its thing. Change it to a list comprehension and it will work.

Reply:
Updated 👍